src/widgets/geo-search/geo-search.js

```diff
@@ -181,18 +182,23 @@ const geoSearch = ({
   const containerNode = getContainerNode(container);
 
   const cssClasses = {
-    root: cx(bem(null), userCssClasses.root),
-    map: cx(bem('map'), userCssClasses.map),
-    controls: cx(bem('controls'), userCssClasses.controls),
-    clear: cx(bem('clear'), userCssClasses.clear),
-    control: cx(bem('control'), userCssClasses.control),
-    toggleLabel: cx(bem('toggle-label'), userCssClasses.toggleLabel),
-    toggleLabelActive: cx(
-      bem('toggle-label-active'),
-      userCssClasses.toggleLabelActive
+    root: cx(suit(), userCssClasses.root),
+    // Required only to mount / unmount the Preact tree
+    tree: suit({ descendantName: 'tree' }),
```

Comment:
Can you explain how this works?

It affects the DOM of the widget. Should we update the specs?

```html
<div class="ais-GeoSearch">
  <div class="ais-GeoSearch-map">
    <!-- Map element here -->
  </div>
  <div class="ais-GeoSearch-tree">
    <div>
      <div class="ais-GeoSearch-control">
        <label class="ais-GeoSearch-label">
          <input class="ais-GeoSearch-input" type="checkbox">
          Search as I move the map
        </label>
      </div>
      <button class="ais-GeoSearch-reset">
        Clear the map refinement
      </button>
    </div>
  </div>
</div>
```

Reply:
The Geo widget use the raw Google Maps to render the map but the rest is rendered with Preact. The `ais-GeoSearch-tree` is required to mount the  Preact root element. We can't use a blank element because inside [the unmount function](https://github.com/algolia/instantsearch.js/blob/ef731cc48c7640aac71e09db799b476ef2474f3b/src/widgets/geo-search/geo-search.js#L252-L254) we have to find this element to unmount the Preact tree with `unmountComponentAtNode`. Since we don't have support of `Fragments` in Preact we have the extra `div` above `ais-GeoSearch-control`. 

I don't think we have to update the specs it's an implementation detail of this widget. React does not have this extra element for example.

Reply:
Thanks for the explanation, this is fine with me.

We need to keep in mind that if ever a user is having trouble styling a node with a CSS selector like `.ais-GeoSearch > .ais-GeoSearch-reset`, it won't work as the specs states.